src/sources/kafka.rs

```diff
@@ -611,13 +612,15 @@ impl ConsumerStateInner<Consuming> {
                             _ => emit!(KafkaReadError { error }),
                         },
                         Some(Ok(msg)) => {
+                            let e = span.enter();
```

Comment:
Would it work to move this `enter` to the top of the `async move` block so that it covers the call to `parse_message` below as well? Alternatively, I think the entire `async_move` block could have a `.instrument(span)` added to it in the call to `spawn`.

Reply:
Yes, good call; instrumenting the entire `async move` block resolves this a bit more straightforwardly :+1: 
I also moved the span into the consumer's inner state struct to make Clippy happy about the argument list length